third_party/packages/ansi_up/lib/ansi_up.dart

```diff
@@ -2,76 +2,306 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// ignore_for_file: avoid_classes_with_only_static_members
-// ignore_for_file: camel_case_types
-// ignore_for_file: non_constant_identifier_names
-// ignore_for_file: prefer_const_declarations
-
 /// ansi_up is an library that parses text containing ANSI color escape
 /// codes.
-@JS()
-library ansi_up;
-
-import 'package:js/js.dart';
 
-@JS()
 class AnsiUp {
-  external AnsiUp();
+  String version, _buffer;
+  bool bold;
+  List<List<AnsiUpColor>> ansiColors;
+  List<AnsiUpColor> palette256;
+  AnsiUpColor fg;
+  AnsiUpColor bg;
+  RegExp _csiRegex;
 
-  external TextPacket get_next_packet();
+  // ignore: sort_constructors_first
+  AnsiUp() {
+    version = '4.0.3';
+    _setupPalettes();
+    bold = false;
+    fg = bg = null;
+    _buffer = '';
+  }
 
-  external void append_buffer(String text);
+  void _setupPalettes() {
+    ansiColors = [
+      [
+        AnsiUpColor(rgb: [0, 0, 0], class_name: 'ansi-black'),
+        AnsiUpColor(rgb: [187, 0, 0], class_name: 'ansi-red'),
+        AnsiUpColor(rgb: [0, 187, 0], class_name: 'ansi-green'),
+        AnsiUpColor(rgb: [187, 187, 0], class_name: 'ansi-yellow'),
+        AnsiUpColor(rgb: [0, 0, 187], class_name: 'ansi-blue'),
+        AnsiUpColor(rgb: [187, 0, 187], class_name: 'ansi-magenta'),
+        AnsiUpColor(rgb: [0, 187, 187], class_name: 'ansi-cyan'),
+        AnsiUpColor(rgb: [255, 255, 255], class_name: 'ansi-white'),
+      ],
+      [
+        AnsiUpColor(rgb: [85, 85, 85], class_name: 'ansi-bright-black'),
+        AnsiUpColor(rgb: [255, 85, 85], class_name: 'ansi-bright-red'),
+        AnsiUpColor(rgb: [0, 255, 0], class_name: 'ansi-bright-green'),
+        AnsiUpColor(rgb: [255, 255, 85], class_name: 'ansi-bright-yellow'),
+        AnsiUpColor(rgb: [85, 85, 255], class_name: 'ansi-bright-blue'),
+        AnsiUpColor(rgb: [255, 85, 255], class_name: 'ansi-bright-magenta'),
+        AnsiUpColor(rgb: [85, 255, 255], class_name: 'ansi-bright-cyan'),
+        AnsiUpColor(rgb: [255, 255, 255], class_name: 'ansi-bright-white'),
+      ]
+    ];
+    palette256 = [];
+    for (var palette in ansiColors) {
+      palette.forEach(palette256.add);
+    }
+    var levels = [0, 95, 135, 175, 215, 255];
+    for (var r = 0; r < 6; ++r) {
+      for (var g = 0; g < 6; ++g) {
+        for (var b = 0; b < 6; ++b) {
+          palette256.add(AnsiUpColor(
+              rgb: [levels[r], levels[g], levels[b]], class_name: 'truecolor'));
+        }
+      }
+    }
+    var greyLevel = 8;
+    for (var i = 0; i < 24; ++i, greyLevel += 10) {
+      palette256.add(AnsiUpColor(
+          rgb: [greyLevel, greyLevel, greyLevel], class_name: 'truecolor'));
+    }
+  }
 
-  external void process_ansi(TextPacket textPacket);
-  external TextWithAttr with_state(TextPacket packet);
+  TextPacket _getNextPacket() {
+    final TextPacket pkt = TextPacket(kind: PacketKind.EOS, text: '', url: '');
```

Comment:
no need to specify the type on the lhs in these two cases as it is inferred.

Reply:
You mean for the `TextPacket` because it's being initialized on the right side, and `int` on the next line because `_buffer.length` is clearly an int? Does this apply to `_buffer.indexOf` below too?